### PR TITLE
Generate contact fields from appointment request reasoncode.text

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- This is part of the api consolidation effort and moves the logic for extracting contact information from the reason code text from the FE to BE. The source logic is [here](https://github.com/department-of-veterans-affairs/vets-website/blob/347cc15aed8c9bfdb857871ab7cde80ff5cb4c05/src/applications/vaos/services/appointment/transformers.js#L41-L60)
- This is work by the Appointments team.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/86895

## Testing done

- [x] *New code is covered by unit tests*
- The old behavior returns the information as part of the reason code text and contact field is empty.
- Tested API response to ensure it matches with ACs

## Screenshots
N/A

## What areas of the site does it impact?
N/A

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Changes ready for review but will continue doing some additional testing.

**Comment by stevenjcumming** [comment]
@JunTaoLuo can you get a teammate to review/approve then I'll review/approve

**Comment by JunTaoLuo** [comment]
@stevenjcumming Of course! @simiadebowale will likely take a look at this PR tomorrow when he's back.

File: modules/vaos/app/services/vaos/v2/appointments_reason_code_service.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module VAOS
+  module V2
+    # Finds the first provider npi id found in provider list and uses that id to request provider name.
```
Comment (simiadebowale) [inline modules/vaos/app/services/vaos/v2/appointments_reason_code_service.rb:5]: Update description.

**Review by jperk51 — APPROVED** [review]
lgtm

File: modules/vaos/app/services/vaos/v2/appointments_service.rb [hunk 2]
```diff
@@ -95,6 +96,7 @@ def post_appointment(request_object_body)
           new_appointment = response.body
           convert_appointment_time(new_appointment)
           find_and_merge_provider_name(new_appointment) if new_appointment[:kind] == 'cc'
+          reason_code_service.extract_reason_code_fields(new_appointment)
```
Comment (simiadebowale) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:99]: The scope of this ticket is to only extract the reason code field for VA appointment requests if appointment.kind == 'cc', we shouldn't extract reasonCode

**Review by simiadebowale — CHANGES_REQUESTED** [review]

**Review by kpethtel — COMMENTED** [review]
@jperk51 We should probably open a ticket for removing this same functionality from our appointments adapter and using the data that's set here instead. But mobile uses reason code text to set more values than this does. We use it for phone, email, preferred dates, reason code, and comment. Does anyone know why mobile and web are using this data differently? It seems like it may result in more discrepancies.

**Comment by simiadebowale** [comment]
@kpethtel - this ticket just covers extracting contact info. We have other tickets to add the rest

File: modules/vaos/spec/services/v2/appointments_reason_code_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe VAOS::V2::AppointmentsReasonCodeService do
+  describe '#extract_reason_code_fields' do
+    it 'returns without modification if no reason code text exists' do
+      appt = FactoryBot.build(:appointment_form_v2, :va_booked_base).attributes
+      subject.send(:extract_reason_code_fields, appt)
```
[the remark this answers is not in the document]
Comment (JunTaoLuo) [reply modules/vaos/spec/services/v2/appointments_reason_code_service_spec.rb:9]: That should work too, I think I was justing using another test as a reference that used this style when I was writing this test.

[the remark this answers is not in the document]
Comment (JunTaoLuo) [reply modules/vaos/spec/services/v2/appointments_reason_code_service_spec.rb:9]: Hmm I tried it and it doesn't seem to work. The test complains the wrong number of arguments is sent so I'll just keep it the way it is now.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb (hunk 2 above)
Comment (JunTaoLuo) [reply modules/vaos/app/services/vaos/v2/appointments_service.rb:99]: Updated the logic in `extract_reason_code_fields`

**Review by simiadebowale — APPROVED** [review]
LGTM

**Pull request merged**